context/src/main/java/io/micronaut/scheduling/TaskScheduler.java

```diff
@@ -59,6 +59,39 @@
      */
     <V> ScheduledFuture<V> schedule(String cron, Callable<V> command);
 
+    /**
+     * Creates and executes a one-shot action that becomes enabled
+     * after the given delay.
+     *
+     * @param cron     The cron expression
+     * @param zoneId The zoneId to base the cron expression on. Defaults to system time zone
+     * @param command  the task to execute
+     * @return a ScheduledFuture representing pending completion of
+     * the task and whose {@code get()} method will return
+     * {@code null} upon completion
+     * @throws java.util.concurrent.RejectedExecutionException if the task cannot be
+     *                                                         scheduled for execution
+     * @throws NullPointerException                            if command or delay is null
+     */
+    ScheduledFuture<?> schedule(String cron, String zoneId, Runnable command);
```

Comment:
These new methods need to be default methods. We cannot add methods to a public interface. The default implementation can just throw an UnsupportedOperationException